File: dataset/codenet/Python/Code/s575946485.py
a = [int(input()) for i in range(5)]

dic = {}

for i in range(5):
    if str(a[i])[-1] == "0":
        dic.update({i: 0})
    else:
        dic.update({i: int(str(a[i])[-1]) - 10})


min_num = [i for i in dic if dic[i] == min(dic.values())][0]
a_sum = sum(a)
dic.pop(min_num)
print(abs(sum(dic.values())) + a_sum)

File: output/deepseek_chat/codenet/direct_translation/python/java/s575946485.java
import java.util.*;

public class s575946485 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int[] a = new int[5];
        for (int i = 0; i < 5; i++) {
            a[i] = scanner.nextInt();
        }
        Map<Integer, Integer> dic = new HashMap<>();
        for (int i = 0; i < 5; i++) {
            String lastDigit = String.valueOf(a[i]).substring(String.valueOf(a[i]).length() - 1);
            if (lastDigit.equals("0")) {
                dic.put(i, 0);
            } else {
                dic.put(i, Integer.parseInt(lastDigit) - 10);
            }
        }
        int min_num = dic.entrySet().stream().min(Map.Entry.comparingByValue()).get().getKey();
        int a_sum = Arrays.stream(a).sum();
        dic.remove(min_num);
        System.out.println(Math.abs(dic.values().stream().mapToInt(Integer::intValue).sum()) + a_sum);
    }
}
